File: BOTAIML.VisionBot.Monitoring.Web/ClientApp/src/app/api/models/role-view-model.ts
/* tslint:disable */
/* eslint-disable */
import { ApplicationPermissions } from './application-permissions';
export interface RoleViewModel {
  id?: number;
  name?: null | string;
  permissions?: null | Array<ApplicationPermissions>;
  reportPermissions?: null | Array<number>;
  users?: null | Array<string>;
}

File: BOTAIML.VisionBot.Monitoring.Web/ViewModels/RoleViewModel.cs
using BOTAIML.VisionBot.Monitoring.Web.Authorization;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace BOTAIML.VisionBot.Monitoring.Web.ViewModels
{
    public class RoleViewModel
    {

        public int Id { get; set; }

        public string Name { get; set; }

        public ApplicationPermissions[] Permissions { get; set; }

        public int[] ReportPermissions { get; set; }

        public IEnumerable<string> Users { get; set; }
    }

    public class UserInfo
    {
        public int Id { get; set; }

        public string Name { get; set; }
    }

}
